Makefile

```diff
@@ -18,9 +18,10 @@ test-fix: vendor
 update-compatibility-patch:
 	@git apply tests/php-compatibility.patch
 	@echo -e "Please open your editor and apply your changes\n"
-	@until [ "$${compatibility_resolved}" == "y" ]; do read -p "Have finished your changes (y|n)? " compatibility_resolved; done && compatibility_resolved=
+	@$$EDITOR tests/expected_report.txt
 	@git diff -- tests/expected_report.txt tests/fixed > .tmp-patch && mv .tmp-patch tests/php-compatibility.patch && git apply -R tests/php-compatibility.patch
 	@git commit -m 'Update compatibility patch' tests/php-compatibility.patch
 
 vendor: composer.json
 	composer update
+	touch vendor
```

Comment:
What about this, why are we assuming composer update won't create vendor folder?

Reply:
We are not assuming it won't create the vendor folder, but we are assuming (righfully) that it will not change its modification date if it already exists.

Reply:
So this updates timestamp of vendor folder, which will make sure dependant targets like `test-report` will run even when vendor wasn't changed 👍

Reply:
If we only want to modify the date, should we use `touch -c` instead? Additionally, are we interested in the directory itself or in all its files (something like `$(shell find vendor -type f)`)? That way, we wouldn't have to touch the directory.

Reply:
If we only want to modify the date, should we use touch -c instead?

@ostrolucky or you @morozov if you use Mac OS can you test `touch -c` or better, `touch --no-create` with `sh` to see if it is portable? I have no idea.

>  Additionally, are we interested in the directory itself or in all its files (something like $(shell find vendor -type f))? That way, we wouldn't have to touch the directory.

What are you suggesting here? `make` only checks the `vendor` directory's timestamp (because it is the target). It will not check any of the underlying files.

Reply:
> @morozov if you use Mac OS

That sounds insulting. No.

> What are you suggesting here?

I meant it to be used in the dependency context e.g. `target: foo bar $(shell  find vendor -type f))`. But I'm not really sure how it's used here, so nevermind.

Reply:
The contents of the `vendor` directory only depends on 2 files `composer.json` and `composer.lock`. And by depends, `make` means they are read (by Composer in this case) when generating that directory.

Reply:
@ostrolucky ran a quick test `--no-create` is not supported, but `-c` is, let's go with that.